lib/net/tns/connection.rb

```diff
@@ -75,11 +76,15 @@ def connect(opts={})
         end
 
         response = send_and_receive(connect_packet)
-        unless response.is_a?(AcceptPacket)
+        unless response.is_a?(AcceptPacket) || response.is_a?(RedirectPacket)
           raise Exceptions::ProtocolException.new("Unexpected response to Connect packet: #{response.class}")
         end
-
+        if response.is_a?(RedirectPacket)
+          # CLR extproc on 12c will end up here
+          return
+        end
```

Comment:
If memory serves, 9i servers (or maybe just the ones I had in the lab) send redirects as well. Not throwing a ProtocolException is definitely an improvement, but we should probably go even further in case a client wanted to handle the redirect. I created #23 for this.